request/extractor.go

```diff
@@ -90,7 +90,7 @@ func (e BearerExtractor) ExtractToken(req *http.Request) (string, error) {
 	tokenHeader := req.Header.Get("Authorization")
 	// The usual convention is for "Bearer" to be title-cased. However, there's no
 	// strict rule around this, and it's best to follow the robustness principle here.
-	if tokenHeader == "" || !strings.HasPrefix(strings.ToLower(tokenHeader), "bearer ") {
+	if tokenHeader == "" || !strings.HasPrefix(strings.ToLower(tokenHeader[:7]), "bearer ") {
```

Comment:
Don't we need a bounds check here, otherwise this will panic? How about:

```go
// getToken extracts the token from the request header. If the token is
// invalid, it returns an empty string.
func getToken(r *http.Request) string {
	tokenHeader := r.Header.Get("Authorization")
	if len(tokenHeader) > 7 && strings.ToLower(tokenHeader[0:6]) == "bearer" {
		return tokenHeader[7:]
	}
	return ""
}
```

Reply:
Fixed it.